test/lib/next-test-utils.ts

```diff
@@ -480,38 +555,29 @@ export async function startApp(app) {
   await app.prepare()
   const handler = app.getRequestHandler()
   const server = http.createServer(handler)
-  server.__app = app
+  server['__app'] = app
+
+  await promisify(server.listen).apply(server)
 
-  await promiseCall(server, 'listen')
   return server
 }
 
-export async function stopApp(server) {
-  if (server.__app) {
-    await server.__app.close()
+export async function stopApp(server: http.Server) {
+  if (server['__app']) {
+    await server['__app'].close()
   }
-  await promiseCall(server, 'close')
-}
-
-export function promiseCall(obj, method, ...args) {
-  return new Promise((resolve, reject) => {
-    const newArgs = [
-      ...args,
-      function (err, res) {
-        if (err) return reject(err)
-        resolve(res)
-      },
-    ]
-
-    obj[method](...newArgs)
-  })
+  await promisify(server.close).apply(server)
```

Comment:
Couldn't figure out how to type `promiseCall` and it doesn't really seem that great anyways